freedomofpress/securedrop-proxy
# Fixes #57 runs black & isor for code formatting check

## Ready for review

Fixes #57. 

Adds new lint targets for `black` and `isort` tool. 

`make lint` will now first check any `isort` change and then `black`
and then `flake8`. All 3 commands use line length of `100`.

The last commit also adds a `.git-blame-ignore-revs` file,
so that we don't see the changes from the formatting commit in the 
`git blame` output.

## How to test manually?

- [ ] go back to the first commit in this PR `git checkout f9bcbd3e8b49b737b2b85bf608f48349ddc68a4b`
- [ ] `make black`
- [ ] `make isort`

All of the above make commands should fail as formatting change is required.

- [ ] move to next commit `git checkout 787844dc339e090aa0e9ac2241895365522c4119`
- [ ] `make black`
- [ ] `make isort`

All of the make commands should now pass. Please note down a few lines in the different files changed in this commit for the next test. Use `git blame` to get the details.

- [ ] move to the latest commit `git checkout 32e62719148d54d1c264e8b781056803fa46b24a`
- [ ] `git config blame.ignoreRevsFile .git-blame-ignore-revs`  run this command.
- [ ] `git blame` should not my name in the lines changed in the last commit. 


# extra notes

Here is one fail message from `black`.

```
$ make lint
would reformat /home/kdas/code/securedrop-proxy/securedrop_proxy/proxy.py
Oh no! 💥 💔 💥
1 file would be reformatted, 8 files would be left unchanged.
make: *** [Makefile:32: black] Error 1
```
